core/src/client/mod.rs

```diff
@@ -213,18 +218,42 @@ pub enum SubscriptionKind {
 	Method(String),
 }
 
-/// Active subscription on the client.
-///
-/// It will try to `unsubscribe` in the drop implementation
-/// but it may fail if the underlying buffer is full.
-/// Thus, if you want to ensure it's actually unsubscribed then
-/// [`Subscription::unsubscribe`] is recommended to use.
+/// Represent a client-side subscription which is implemented on top of 
+/// a bounded channel where it's possible that the receiver may
+/// not keep it with the sender side a.k.a "slow receiver problem"
+/// 
+/// ## Lagging
+/// 
+/// As all sent messages must be kept in a buffer until the underlying
+/// stream is polled, it's possible that the server is producing
+/// subscription notifications faster than the client can handle.
```

Comment:
Just to check that I understand this lagging stuff; I think the idea is that the server is sending messages to the client, but if you don't poll the client enough then the buffer on the client will fill up, and then we'll start missing messages if we aren't polling the client subscription enough to drain the buffer right?

```suggestion
/// All messages from the server must be kept in a buffer in the client
/// until they are read by polling the [`Subscription`]. If you don't 
/// poll the client subscription quickly enough, the buffer may fill 
/// up, which will result in messages being lost.
```